src/riak_kv_replrtq_snk.erl

```diff
@@ -409,13 +428,45 @@ handle_info({prompt_requeue, WorkItem}, State) ->
     requeue_work(WorkItem),
     {noreply, State}.
 
-terminate(_Reason, _State) ->
+terminate(_Reason, State) ->
+    WorkItems = lists:map(fun(SW) -> element(3, SW) end, State#state.work),
+    CloseFun = 
+        fun(SinkWork) ->
+            lists:foreach(
+                fun({{_QN, _Iter, _Peer}, _LocalC, RemoteFun, _RCF}) ->
+                    RemoteFun(close)
+                end,
+                SinkWork#sink_work.work_queue)
+        end,
+    lists:foreach(CloseFun, WorkItems),
     ok.
 
 code_change(_OldVsn, State, _Extra) ->
     {ok, State}.
 
 
+%%%============================================================================
+%%% External functions
+%%%============================================================================
+
+get_worker_counts() ->
+    SnkWorkerCount =
+        app_helper:get_env(riak_kv,
+                            replrtq_sinkworkers,
+                            ?DEFAULT_WORKERCOUNT),
+    PerPeerLimit =
+        app_helper:get_env(riak_kv,
+                                replrtq_sinkpeerlimit,
+                                SnkWorkerCount),
+    {SnkWorkerCount, PerPeerLimit}.
+
+set_worker_counts(SnkWorkerCount, PerPeerLimit) ->
+    application:set_env(riak_kv, replrtq_sinkworkers, SnkWorkerCount),
```

Comment:
Why using app_helper for getting and application for setting? Is this an artefcat of OTP18 or so?

Reply:
Never thought about this much, just following inline with its use elsewhere.  Looking at the docs application:get_env/3 didn't exist until R16B ... and Riak was initially written prior to that ... so I guess it is just a legacy of this.

It might be one thing to add to the list for the OTP 24+ version of Riak, to refactor throughout and to take this out.